plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/worker/KtLintInvocation.kt

```diff
@@ -0,0 +1,351 @@
+    val editorConfigOverride = editorConfigOverrideClass.kotlin.primaryConstructor!!.call()
+    val addMethod = editorConfigOverrideClass.getDeclaredMethod("add", getEditorConfigPropertyClass(), Any::class.java)
+    addMethod.isAccessible = true
+    val disabledRulesProperty =
+        defaultEditorConfigPropertiesClass.kotlin.memberProperties.firstOrNull { it.name == "ktlintDisabledRulesProperty" }
+            ?: defaultEditorConfigPropertiesClass.kotlin.memberProperties.first { it.name == "disabledRulesProperty" }
+    val codeStyleSetProperty =
+        defaultEditorConfigPropertiesClass.kotlin.memberProperties.first { it.name == "codeStyleSetProperty" }
+    addMethod.invoke(
+        editorConfigOverride, disabledRulesProperty.getter.call(defaultEditorConfigProperties),
+        userData["disabled_rules"]
+    )
+    addMethod.invoke(editorConfigOverride, codeStyleSetProperty.getter.call(defaultEditorConfigProperties), codeStyle)
+    return editorConfigOverride
+}
+
+/**
+ * Implementation for invoking ktlint 0.47.x
+ */
+@OptIn(FeatureInAlphaState::class)
+internal class ExperimentalParamsProviderInvocation(
+    private val editorConfigPath: String?,
+    private val ruleProviders: Set<Any>,
+    private val userData: Map<String, String>,
+    private val debug: Boolean,
+) : KtLintInvocation {
+    companion object Factory : KtLintInvocationFactory {
+        fun initialize(
+            editorConfigPath: String?,
+            ruleProviders: Set<Any>,
+            userData: Map<String, String>,
+            debug: Boolean
+        ): ExperimentalParamsProviderInvocation =
+            ExperimentalParamsProviderInvocation(editorConfigPath, ruleProviders, userData, debug)
+    }
+
+    private val ctor: KFunction<*> by lazy {
+        experimentalParamsClass?.kotlin?.primaryConstructor!!
+    }
+    private val fileNameParam: KParameter by lazy { ctor.findParameterByName("fileName")!! }
+    private val textParam: KParameter by lazy { ctor.findParameterByName("text")!! }
+    private val ruleProvidersParam: KParameter by lazy { ctor.findParameterByName("ruleProviders")!! }
+    private val cbParam: KParameter by lazy { ctor.findParameterByName("cb")!! }
+    private val scriptParam: KParameter by lazy { ctor.findParameterByName("script")!! }
+    private val editorConfigPathParam: KParameter by lazy { ctor.findParameterByName("editorConfigPath")!! }
+    private val debugParam: KParameter by lazy { ctor.findParameterByName("debug")!! }
+    private val editorConfigOverrideParam: KParameter by lazy { ctor.findParameterByName("editorConfigOverride")!! }
+
+    private fun buildParams(
+        file: File,
+        cb: (LintError, Boolean) -> Unit
+    ): com.pinterest.ktlint.core.KtLint.ExperimentalParams {
+        val script = !file.name.endsWith(".kt", ignoreCase = true)
+        val editorConfigOverride = userDataToEditorConfigOverride(userData)
```

Comment:
Store this in a local lazy variable. It doesn't need to be created for every file